src/sentry/lang/java/plugin.py

```diff
@@ -125,6 +127,136 @@ def process_frame(self, processable_frame, processing_task):
         return
 
 
+# A processor that delegates to JavaStacktraceProcessor for restoring code
+# obfuscated by ProGuard or similar. It then tries to look up source context
+# for either the de-obfuscated stack frame or the stack frame that was passed in.
+class JavaSourceLookupStacktraceProcessor(StacktraceProcessor):
+    def __init__(self, *args, **kwargs):
+        StacktraceProcessor.__init__(self, *args, **kwargs)
+        self.proguard_processor = JavaStacktraceProcessor(*args, **kwargs)
+        self._proguard_processor_handles_frame = None
+        self._handles_frame = None
+        self.images = get_jvm_images(self.data)
+        self.available = len(self.images) > 0
+
+    def handles_frame(self, frame, stacktrace_info):
+        self._proguard_processor_handles_frame = self.proguard_processor.handles_frame(
+            frame, stacktrace_info
+        )
+
+        platform = frame.get("platform") or self.data.get("platform")
+        self._handles_frame = (
+            platform == "java" and self.available and "module" in frame and "lineno" in frame
```

Comment:
I assumed that even with proguard if there's no line number in the first place we'd be unable to produce one via proguard mapping. No `lineno` means no source lookup possible. Maybe my assumption is wrong. I can remove the `lineno` check. Though the check here doesn't mean we use the "raw" `lineno` for lookup. In case `JavaStacktraceProcessor` (doing proguard mapping) remaps it, we take the remapped `lineno`. This behaviour is also tested by the last test.